index.js

```diff
@@ -0,0 +1,203 @@
+      { NAME: "Raisins", AMOUNT: "100 g" },
+      { NAME: "Slipped Almonds", AMOUNT: "50 g" },
+    ],
+    preparationSteps: ["Mix everything, heat it, and you are good to go!"],
+  },
+];
+
+function setRecipeTitle(recipe) {
+  const recipeTitle = document.getElementById("recipe-title");
+  if (recipeTitle) {
+    recipeTitle.textContent = recipe.title;
+  } else {
+    console.error("Element #recipe-title not found");
+  }
+}
+
+function setRecipeImage(recipe) {
+  const recipeImage = document.getElementById("recipe-image");
+  if (recipeImage) {
+    recipeImage.src = recipe.picture_url;
+    recipeImage.alt = recipe.title;
+  } else {
+    console.error("Element #recipe-image not found");
+  }
+}
+
+function setRecipeDescription(recipe) {
+  const recipeDescription = document.getElementById("recipe-description");
+  if (recipeDescription) {
+    recipeDescription.innerText = recipe.description;
+  } else {
+    console.error("Element #recipe-description not found");
+  }
+}
+
+function setIngredients(recipe) {
+  const ingredientList = document.querySelector("#recipe-ingredients ul");
+  if (ingredientList) {
+    ingredientList.innerHTML = "";
+    recipe.ingredients.forEach((ingredient) => {
+      const li = document.createElement("li");
+      if (ingredient.NAME && ingredient.AMOUNT) {
+        li.textContent = `${ingredient.NAME}: ${ingredient.AMOUNT}`;
+      } else if (ingredient.NAME && !ingredient.AMOUNT) {
+        li.textContent = `${ingredient.NAME}: appropriate`;
+      }
+      ingredientList.appendChild(li);
+    });
+  } else {
+    console.log("Element #recipe-ingredients ul not found");
+  }
+}
+
+function setPreparationSteps(recipe) {
+  const preparationStepsList = document.querySelector("#preparation ul");
+  preparationStepsList.innerHTML = "";
+  if (preparationStepsList) {
+    recipe.preparationSteps.forEach((step) => {
+      const li = document.createElement("li");
+      li.textContent = step;
+      preparationStepsList.appendChild(li);
+    });
+  } else {
+    console.error("Element #preparation ul not found");
+  }
+}
+
+
+
+const newIngredientForm = document.getElementById("new-ingredient-form");
+if (newIngredientForm) {
+  newIngredientForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+
+    const newIngredientName = document.getElementById("ingredient-name").value;
+    const newIngredientAmount =
+      document.getElementById("ingredient-amount").value;
+
+    if (!newIngredientName) {
+      alert("Please provide ingredient name.");
+      return;
+    }
+
+    const currentRecipe = recipeObject[recipeObject.length - 1];
+    currentRecipe.ingredients.push({
+      NAME: newIngredientName,
+      AMOUNT: newIngredientAmount,
+    });
+
+    document.getElementById("ingredient-name").value = "";
+    document.getElementById("ingredient-amount").value = "";
+
+    setIngredients(currentRecipe);
+  });
+} else {
+  console.error("Element #new-recipe-form not found");
+}
```

Comment:
Some users are complaining that for simple ingredients like salt/pepper they may not want/know the exact amount. Should/can we fix that?